packages/instrumentation-amqplib/test/amqplib-promise.spec.ts

```diff
@@ -0,0 +1,514 @@
+    const memoryExporter = new InMemorySpanExporter();
+    const spanProcessor = new SimpleSpanProcessor(memoryExporter);
+    propagation.setGlobalPropagator(new HttpTraceContext());
+    provider.addSpanProcessor(spanProcessor);
+    instrumentation.setTracerProvider(provider);
+
+    const url = 'amqp://localhost:22221';
+    let conn: amqp.Connection;
+    before(async () => {
+        instrumentation.enable();
+        conn = await amqp.connect(url);
+    });
+    after(async () => {
+        await conn.close();
+        instrumentation.disable();
+    });
+
+    let endHookSpy;
+    const expectConsumeEndSpyStatus = (expectedEndOperations: EndOperation[]): void => {
+        expect(endHookSpy.callCount).toBe(expectedEndOperations.length);
+        expectedEndOperations.forEach((endOperation: EndOperation, index: number) => {
+            expect(endHookSpy.args[index][3]).toStrictEqual(endOperation);
+            switch (endOperation) {
+                case EndOperation.AutoAck:
+                case EndOperation.Ack:
+                case EndOperation.AckAll:
+                    expect(endHookSpy.args[index][2]).toBeFalsy();
+                    break;
+
+                case EndOperation.Reject:
+                case EndOperation.Nack:
+                case EndOperation.NackAll:
+                case EndOperation.ChannelClosed:
+                case EndOperation.ChannelError:
+                    expect(endHookSpy.args[index][2]).toBeTruthy();
+                    break;
+            }
+        });
+    };
+
+    let channel: amqp.Channel;
+    beforeEach(async () => {
+        memoryExporter.reset();
+
+        // instrumentation.disable();
+        endHookSpy = sinon.spy();
+        instrumentation.setConfig({
+            consumeEndHook: endHookSpy,
+        });
+        instrumentation.enable();
+
+        channel = await conn.createChannel();
+        await channel.assertQueue(queueName, { durable: false });
+        await channel.purgeQueue(queueName);
+        // install an error handler, otherwise when we have tests that create error on the channel,
+        // it throws and crash process
+        channel.on('error', () => {});
+    });
+    afterEach(async () => {
+        try {
+            channel.close();
+        } catch {}
+        instrumentation.disable();
+    });
+
+    it('simple publish and consume from queue', async () => {
+        const hadSpaceInBuffer = channel.sendToQueue(queueName, Buffer.from(msgPayload));
+        expect(hadSpaceInBuffer).toBeTruthy();
+
+        await asyncConsume(channel, queueName, [(msg) => expect(msg.content.toString()).toStrictEqual(msgPayload)], {
+            noAck: true,
+        });
+        const [publishSpan, consumeSpan] = memoryExporter.getFinishedSpans();
+
+        // assert publish span
+        expect(publishSpan.kind).toStrictEqual(SpanKind.PRODUCER);
+        expect(publishSpan.attributes[MessagingAttribute.MESSAGING_SYSTEM]).toStrictEqual('rabbitmq');
```

Comment:
not very important but no point using toStrictEqual when comparing strings, it's mostly for objects.
https://jestjs.io/docs/expect#tostrictequalvalue

Reply:
cool, didn't kwno the difference. changed to `toEqual`